src/Microsoft.AspNetCore.Mvc.Core/RequestSizeLimitAttribute.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Microsoft.AspNetCore.Mvc.Core
+{
+    /// <summary>
+    /// A filter that specifies the request body size limit.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class RequestSizeLimitAttribute : Attribute, IRequestSizePolicy, IOrderedFilter, IResourceFilter
+    {
+        private long _bytes;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="RequestSizeLimitAttribute"/>.
+        /// </summary>
+        public RequestSizeLimitAttribute(long bytes)
+        {
+            _bytes = bytes;
+        }
+
+        /// <inheritdoc />
+        public bool IsReusable => true;
+
+        /// <inheritdoc />
+        public int Order { get; set; } = 1000;
+
+        /// <inheritdoc />
+        public void OnResourceExecuted(ResourceExecutedContext context)
+        {
+
+        }
+
+        /// <summary>
+        /// As an <see cref="IResourceFilter"/>, this filter looks at the request and rejects it before going ahead if
+        /// the request body size is greater than the specified limit.
+        /// </summary>
+        /// <param name="context">The <see cref="ResourceExecutingContext"/>.</param>
+        public void OnResourceExecuting(ResourceExecutingContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (IsClosestRequestSizePolicy(context.Filters))
+            {
+                var maxRequestBodySizeFeature = context.HttpContext.Features.Get<IHttpMaxRequestBodySizeFeature>();
+                maxRequestBodySizeFeature.MaxRequestBodySize = _bytes;
```

Comment:
Same comments apply here, we need to check for null.

@Tratcher - any ideas on what a reasonable behavior is when a server doesn't support this feature?

I could see throwing, but the problem will be that we're going to encourage users to use things like TestServer.